### ISSUE TITLE
Leading './' is not included in the glob regex

### ISSUE THREAD

**Issue by valerybugakov** [micromatch/micromatch#199]
## **Minimum necessary steps** to reproduce this issue:

```js
const testGlob = '**/*.js';

console.log(micromatch.isMatch('./something/file.js', testGlob));
console.log(micromatch.isMatch('something/file.js', testGlob));
console.log(micromatch.makeRe(testGlob).test('./something/file.js'));
console.log(micromatch.makeRe(testGlob).test('something/file.js'));
```

## What is happening that shouldn't be?

Output

```
true
true
false
true
```

## What should be happening instead?

Expected

```
true
true
true
true
```

This issue is the exact duplicate of https://github.com/micromatch/micromatch/issues/110. It was [resolved](https://github.com/micromatch/nanomatch/commit/770f54c59b3a4b60a92dd1c2a396018b5596f13a) in `nanomatch` long ago. It works correctly using `makeRe` from `nanomatch`.

Should it be fixed in the same way in `micromatch`, probably in `picomatch` that is used under the hood?

**Comment by jonschlinkert** [comment]
See the [format option](https://github.com/micromatch/picomatch#options).

**Comment by valerybugakov** [comment]
It solves the problem for `isMatch`.
I cannot figure out how to solve it for regex created via `picomatch.makeRe(glob)`. Is it possible?

```ts
const format = str => str.replace(/^\.\//, '');
const regex = picomatch.makeRe('foo/*.js', { format });
console.log(regex.test('./foo/bar.js')); //=> false but expected true
```

**Comment by jonschlinkert** [comment]
> Is it possible?

Maybe try using the `prepend` option. However, IMHO the easiest and possibly the most reliable way to do this is to remove the `./` from paths before testing them. This is what I'd personally do if I was implementing a globbing library, since 1) it would most likely be faster than a more complicated regex, and 2) file paths do not typically have `./` before them. AFAIK that is only manually added when defining a `require()` path. I see devs add `./` before paths all the time and it doesn't make a lot of sense.

**Comment by valerybugakov** [comment]
Agreed, but I have the same case as described in the issue linked above. 

> I am trying to convert the webpack require-context api into accepting a glob (it takes a regex). The strings it tests against that regex are always relative file paths that start with ./ and I cannot strip this prefix with their available api.

Is it safe to use `nanomatch` in that case? I see it hasn't been update for a while.

**Comment by valerybugakov** [comment]
btw thanks for quick response 🙌

**Comment by jonschlinkert** [comment]
Ah, I'm sorry for not reading more carefully and if my answer seemed to ignore those details. I didn't mean to be flippant. I see that this is more nuanced. Before I implement something new, let me think about this for a few, it seems like this came up before and I/we had a solution to it. 

> Is it safe to use `nanomatch` in that case? I see it hasn't been update for a while.

It's not as performant. I think we should try to make this work with picomatch. I thought the `prepend` option allowed you to add an arbitrary string to the beginning of the regex, which would allow you to add something like `(\\.[\\\\/])?`.

**Comment by valerybugakov** [comment]
Got it, will try the `prepend` option for now.

**Comment by hydrosquall** [comment]
FWIW, `minimatch` has faced this problem for several years too, in case you want to collaborate on a solution. 

https://github.com/isaacs/minimatch/issues/30#issue-29835003

**Comment by dawgwe1** [comment]
Close all accounts

**Comment by jonschlinkert** [comment]
@dawgwe1 you've been blocked
